ffcx/ir/representation.py

```diff
@@ -79,8 +79,9 @@ def compute_ir(analysis, object_names, prefix, parameters, visualise):
     for fd_index, fd in enumerate(analysis.form_data):
         form_names[fd_index] = naming.form_name(fd.original_form, fd_index, prefix)
         for itg_index, itg_data in enumerate(fd.integral_data):
-            integral_names[(fd_index, itg_index)] = naming.integral_name(fd.original_form, itg_data.integral_type,
-                                                                         fd_index, itg_data.subdomain_id, prefix)
+            # Unique ID for integrals is 2**form_index * 3**integral_index
+            integral_names[(fd_index, itg_index)] = naming.integral_name(
+                fd.original_form, itg_data.integral_type, 2**fd_index * 3**itg_index, prefix)
```

Comment:
What is this `2**fd_index * 3**itg_index` for? Are you encoding multiple integers into one using prime factorisation?

Reply:
Yes, as we now only have a single kernel for integrals with multiple subdomain ids, we need to have a way of giving them a unique name based on the kernel and the location in the subdomain_ids.

Reply:
I would suggest to pass all integers that make up unique integral name and hash it all later in the naming.

Reply:
I'm not sure I follow. How would you do this?

Reply:
```suggestion
                fd.original_form, itg_data.integral_type, fd_index, itg_index, prefix)
```
and
```python
def integral_name(original_form, integral_type, form_id, integral_id, prefix):
    sig = compute_signature([original_form], str((prefix, integral_type, form_id, integral_id)))
```